tv_apps/smart-home/js/home.js

```diff
@@ -187,12 +193,77 @@
           cardManager: that.cardManager
         });
         that._cardPicker.on('hide', that.onCardPickerHide.bind(that));
+        that._cardPicker.on('show', that.onCardPickerShow.bind(that));
+      });
+    },
+
+    initContextMenu: function () {
+      this.moveMenuItem.addEventListener('click', () => {
+          this.edit.toggleEditMode();
+          this.edit.once('arranged', () => this.edit.toggleEditMode());
+          this.edit.toggleArrangeMode();
+      });
+
+      this.renameMenuItem.addEventListener('click', () => {
+        var focus = this.spatialNavigator.getFocusedElement();
+        this.edit.toggleEditMode();
+        this.edit.once('rename-card-ended', () => this.edit.toggleEditMode());
+        this.edit.renameCard(focus, focus.getNodeFromItem(focus.currentItem));
+      });
+
+      this.removeMenuItem.addEventListener('click', () => {
+        var focus = this.spatialNavigator.getFocusedElement();
+        this.edit.toggleEditMode();
+        this.edit.once('delete-card-ended', () => this.edit.toggleEditMode());
+        this.edit.openDeleteCardDialog(focus,
+          focus.getNodeFromItem(focus.currentItem));
+      });
```

Comment:
And actually switching to edit mode creates unnecessary animation. Can you try not to switch into edit mode and just remove card directly? Maybe some edit-mode specified css animation rules need to be changed to trigger animationEnd correctly.